### PR TITLE
EzRpc: extend interface to support arbitrary streams

### PR DESCRIPTION
**Pull request by JonasVautherin**
__TL;DR: The whole point is the new API in [ez-rpc.h](https://github.com/capnproto/capnproto/compare/master...JonasVautherin:capnproto:ezrpc-arbitrary-asynciostream?expand=1#diff-548cb29d954120f68618ee55c221ea96b5d73c29a86eefdc1bb2360fba0e851d), check it first, no need to review the rest if the API is bad :+1:.__

This is an exploration around the EzRpc API, to see if it can be extended to support arbitrary streams (enabling e.g. tls, gzip, websocket). It adds:

* An `EzRpcClient` constructor without a stream, which can be set later with `EzRpcClient.setStream()`
* A constructor to create an "empty" `EzRpcServer` (without a stream and without an `acceptLoop`)
* An `EzRpcServer.addStream()` function to add streams to the server object (those streams can come from an external `acceptLoop`).

@kentonv: I really don't want to lose your time with stupid propositions, so feel free to close if this seems useless or convoluted :sweat_smile:.

Note: pretty sure I messed up with the `PromiseAndFulfiller`, happy to get advice there :innocent:.

### CODE REVIEW

**Comment by JonasVautherin** [comment]
Closing, because most likely one of the following statements is true:

    Nobody cares/has time to care (no offense, really, just trying to be realistic)
    It is debatable whether this improves the EzRpc API or not
    I don't need it myself, I was really just trying to improve EzRpc

Also, if there is ever extra resources to review one of my PRs, I would much rather get it for the https://github.com/capnproto/capnproto/pull/1460 innocent.

**Comment by JonasVautherin** [comment]
Closing, because most likely one of the following statements is true:

* Nobody cares/has time to care (no offense, really, just trying to be realistic)
* It is debatable whether this improves the EzRpc API or not
* I don't need it myself, I was really just trying to improve EzRpc

Also, if there is ever extra resources to review one of my PRs, I would much rather get it for the https://github.com/capnproto/capnproto/pull/1460 innocent.

**Comment by kentonv** [comment]
Hi @JonasVautherin,

Sorry for the radio silence, this PR was posted while I was on PTO and ignoring my email. (Yes, I was posting PRs of my own in the meantime, but that's sort of my ideal vacation: writing my own code while ignoring everyone else. :sweat_smile:)

With that said, to be honest I have long felt that EZ RPC was a mistake and I'd prefer to remove it entirely. It hides too much, which creates a barrier for people as soon as they want to do something more complicated. Meanwhile, the "real" APIs are barely any harder to use in practice, so I'd rather direct people to them than try to improve EZ RPC.

I just need to find time to update all the docs...
